tests/test.cpp

```diff
@@ -1,13 +1,322 @@
+
+    //Now it is set
+    ASSERT_EQ(mockTown6.A->getRoadNumber(),1);
+    ASSERT_EQ(mockTown6.B->getRoadNumber(),1);
+
+    //Verify that the Road knows its ends
+    ASSERT_EQ(R.getOther(mockTown6.A->getNodeID()).getNodeID(),mockTown6.B->getNodeID());
+    ASSERT_EQ(R.getOther(mockTown6.B->getNodeID()).getNodeID(),mockTown6.A->getNodeID());
+
+
+    //Using Global ID
+    ASSERT_EQ(mockTown6.A->getRoad(R.getRoadID()).getRoadID(), R.getRoadID());
+    //The one and only road
+    ASSERT_EQ(mockTown6.A->getRoad(0,true).getRoadID(), R.getRoadID());
+
+//Same thing but with the neighbour node
+
+    //Using Global ID
+    ASSERT_EQ(mockTown6.A->getNeighbour(R.getRoadID()).getNodeID(), mockTown6.B->getNodeID());
+    //The one and only road
+    ASSERT_EQ(mockTown6.A->getNeighbour(0,true).getNodeID(), mockTown6.B->getNodeID());
+
+//A few illegal ones
+
+    ASSERT_THROW(mockTown6.A->getRoad(static_cast<size_t>(-1)/*VERY invalid ID*/), road_address_exception);
+    ASSERT_THROW(mockTown6.A->getRoad(mockTown6.A->getMaxRoadNumber()+1/*Not in our local list of neighbours*/,true), road_address_exception);
+
+    ASSERT_THROW(mockTown6.A->getNeighbour(static_cast<size_t>(-1)).getNodeID(), road_address_exception);
+    //The one and only road
+    ASSERT_THROW(mockTown6.A->getNeighbour(mockTown6.A->getMaxRoadNumber()+1,true).getNodeID(), road_address_exception);
+
+
+    //Now do the exact same thing with the other node, just to see that they connect to each other
+
+
+
+    //Using Global ID
+    ASSERT_EQ(mockTown6.B->getRoad(R.getRoadID()).getRoadID(), R.getRoadID());
+    //The one and only road
+    ASSERT_EQ(mockTown6.B->getRoad(0,true).getRoadID(), R.getRoadID());
+
+//Same thing but with the neighbour node
+
+    //Using Global ID
+    ASSERT_EQ(mockTown6.B->getNeighbour(R.getRoadID()).getNodeID(), mockTown6.A->getNodeID());
+    //The one and only road
+    ASSERT_EQ(mockTown6.B->getNeighbour(0,true).getNodeID(), mockTown6.A->getNodeID());
+
+//B few illegal ones
+
+    ASSERT_THROW(mockTown6.B->getRoad(static_cast<size_t>(-1)/*VERY invalid ID*/), road_address_exception);
+    ASSERT_THROW(mockTown6.B->getRoad(mockTown6.B->getMaxRoadNumber()+1/*Not in our local list of neighbours*/,true), road_address_exception);
+
+    ASSERT_THROW(mockTown6.B->getNeighbour(static_cast<size_t>(-1)).getNodeID(), road_address_exception);
+    //The one and only road
+    ASSERT_THROW(mockTown6.B->getNeighbour(mockTown6.B->getMaxRoadNumber()+1,true).getNodeID(), road_address_exception);
+
+
+
+}
+
+
+/*MISSING
+Add a car to a road, advance time and verify that the car
+verify that exceptions are thrown when "overstepping" updates for car
+Adding a car to the road, should have both the car and the road associated with each other.
+The number of cars exiting the road, should match the number of cars entering the road
+At the end, the car should know that it is not on the road anymore.
+The keyframe JSON file must have the expected keyframes
 */
```

Comment:
This will be done in a future branch. It was originally intended here, but that would make this branch too large